lib/jazzy/sourcekitten.rb

```diff
@@ -100,6 +110,33 @@ def self.group_type_categories(docs, type_category_prefix)
       [group.compact, docs]
     end
 
+    def self.navigation_module_section(docs, modules)
+      group = modules.map do |modulename|
+        children, docs = docs.partition { |doc| doc.modulename == modulename }
+        make_group(
+          children,
+          modulename,
+          "",
+        )
+      end
+
+      # Get from the remaining docs if there are extensions that should also be part of this module.
+      group = group.compact.map { |group|
```

Comment:
I get what this is doing; some issues/questions:

1. Using a child decl to identify the declaring module is neat, but not all extensions have children (think `extension SomeType : MyProtocol {}`).
2. When naming `newdoc` how about including the originating module name?  So it would be `MyModule+Swift.Int` or something -- otherwise not sure how the user knows where the extended type is from.  (Or even just `Swift.Int` given this is already nested under `MyModule`)
3. I don't understand the scenario on line 137 where there is stuff left over that doesn't belong to any of the modules.  What is an example of this?


About point 1, maybe think about changing how `parse()` works for multiple modules: could do each module's JSON one by one, then would always know which module is being worked on.  At the moment `group_docs_per_module` is really just *undoing* work that's already been done to combine everything.